### PR TITLE
Remove `page_title` attribute from frontmatter in resource documentation, update external link

### PR DESCRIPTION
**Pull request by SarahFrench**
This PR removes a line from the template used to generated markdown files documenting MMv1 resources. Currently the template adds a `page_title` attribute in the YAML frontmatter of each resource's markdown file, but this field is not required for resource documentation. [The documentation for provider documentation](https://developer.hashicorp.com/terraform/registry/providers/docs#supported-attributes) describes how this attribute is only needed for files in the `/guides` folder. Those files are all handwritten and not related to this template, so it's safe to remove this field from the template.

This PR also updates a link in the `Timeouts` section in the docs ([example here](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/app_engine_service_split_traffic#timeouts)) . The template currently includes a link to the old documentation that was hosted at terraform.io. Although the old link is redirected to the new developer.hashicorp.com website, the docs have changed and the link doesn't send users to a relevant page.

___

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- ~~[x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).~~
- [x] ~~[Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).~~
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/templates/terraform/resource.html.markdown.erb [hunk 1]
```diff
@@ -173,7 +172,7 @@ In addition to the arguments listed above, the following computed attributes are
 ## Timeouts
 
 This resource provides the following
-[Timeouts](/docs/configuration/resources.html#timeouts) configuration options:
+[Timeouts](https://developer.hashicorp.com/terraform/plugin/sdkv2/resources/retries-and-customizable-timeouts) configuration options:
```
Comment (SarahFrench) [inline mmv1/templates/terraform/resource.html.markdown.erb:175]: Can't use the relative link anymore as docs have moved away from the terraform.io domain

**Comment by SarahFrench** [comment]
Realised there are a ton of other templates - marking as draft until I've found them all

**Comment by SarahFrench** [comment]
This PR is similar to [this past PR](https://github.com/GoogleCloudPlatform/magic-modules/pull/6238) that also cleaned up usage of frontmatter in the provider documentation.

Sorry that there are a lot of file changes (here and in the downstream). If it helps with reviewing this PR, the changes to the handwritten files were achieved using a global search and replace, where I searched with this regex (`\npage_title: "Google: google_[a-z_]+"`) and replaced with nothing.

**Review by shuyama1 — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**